### PR TITLE
Issue5

### LINKED ISSUE #5: A system for data prep that will be easy to maintain
Based on our previous discussion, I understand we want `OpenLong` to provide a function for each cohort that can produce the cleaned and harmonized version of the cohort data. This is important for users who have access to the real data, as they should be able to run the `OpenLong` functions and get real, cleaned data as an output.  We should discuss and hopefully come to an agreement about what system would work best for managing this in `OpenLong`. 

I have some thoughts on what could work well. The idea is to write several functions for each cohort:

1. `data_load()` - this function should pull the relevant files out of the package of files that would be delivered from NHLBI BioLINCC. It will need to be customized for each cohort since each cohort comes with a different bundle of files in the delivered data. 
1. `data_clean()` - this function should *modify* existing columns in the dataset to align them with out standard for harmonizing. For example, if a variable initially has values of, e.g., -999, to indicate missingness, this function would set those to `NA_real` or whatever type of `NA` matches the format of that column. Another action taken by this function would be recoding factor levels to match the standard we set for the given variable in our harmonization. E.g., one cohort may have levels of "M" and "F". Assuming we decided to universally code these as 'male' and 'female' we would do the recoding in this function. 
1. `data_derive()` - this function should *create new* columns using the cleaned versions of existing columns.  
1. `data_exclude()` - this function should filter out rows that do not meet inclusion criteria for our harmonized datasets. If we don't plan to have inclusion criteria, then this function isn't needed.  
1. `data_rename()` - this function should map a datasets existing names to the harmonized names. It should just leverage the .xlsx file that Brian prepared. We could make this .xlsx file one of the internal datasets used by `OpenLong` for convenience. 
1. `data_finalize()` - this function should separate a cross-sectional and longitudinal dataset given a 

The reason I feel adamant about clean distinctions b/t the purpose of each function is for our own convenience. When we are asked questions like, "how did you create that variable", we know exactly where to look (in the `data_derive()` function). It also helps us write a checklist out when we start harmonizing a new dataset.  

We don't have to actually implement this system to resolve this issue, we just have to discuss and make a plan.

### PR DESCRIPTION
**Pull request by bcjaeger**
Fixes #5 

This PR provides a minimal S7 system with some classes for ABC and MESA. If this gets pulled into main, we should follow up with creating issues for Brian and Simar to populate code inside the MESA and ABC classes.  

Note that there are TODO items in the code, but those TODOs are not required for this PR to be merged.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,4 +53,4 @@ po/*~
 rsconnect/
 
 # sensitive data
-/sensitive/
+data-raw/sensitive/
```
Comment (bcjaeger) [inline .gitignore:56]: I checked to make sure this works. For some reason /sensitive/ wasn't ignoring data-raw/sensitive the way we wanted it to.

File: DESCRIPTION [hunk 2]
```diff
@@ -15,5 +15,11 @@ Config/testthat/edition: 3
 URL: https://github.com/BrianNathanWhite/OpenLong
 BugReports: https://github.com/BrianNathanWhite/OpenLong/issues
 Depends: 
-    R (>= 2.10)
+    R (>= 4.3.0)
```
Comment (bcjaeger) [inline DESCRIPTION:18]: for ease of use with S7

File: DESCRIPTION [hunk 3]
```diff
@@ -15,5 +15,11 @@ Config/testthat/edition: 3
...
 LazyData: true
+Imports: 
+    dplyr,
+    readr,
+    S7,
+    tibble,
+    tidyr
```
Comment (bcjaeger) [inline DESCRIPTION:25]: this is a hard commit to the tidyverse. We could do data.table instead but I think our team is more tidyverse trained.

**Comment by BrianNathanWhite** [comment]
This all looks good to me! I'll go ahead and merge the pull request.

**Pull request merged**